packages/cli/__tests__/zosjobs/__system__/search/cli.zos-jobs.search.job.test.ts

```diff
@@ -9,98 +9,125 @@
-        TEST_ENVIRONMENT = await TestEnvironment.setUp({
+        // Initialize testEnvironment first!
+        testEnvironment = await TestEnvironment.setUp({
             testName: "zos_jobs_search_job_command",
             tempProfileTypes: ["zosmf"]
-        });
+        }, REAL_SESSION = await TestEnvironment.createSession());
+
+        // Use testEnvironment for accessing properties
+        defaultSystem = testEnvironment.systemTestProperties;
 
-        IEFBR14_JOB = TEST_ENVIRONMENT.systemTestProperties.zosjobs.iefbr14Member;
-        const defaultSystem = TEST_ENVIRONMENT.systemTestProperties;
-        REAL_SESSION = TestEnvironment.createZosmfSession(TEST_ENVIRONMENT);
+        IEFBR14_JOB = defaultSystem.zosjobs.iefbr14Member;
 
         ACCOUNT = defaultSystem.tso.account;
-        const JOB_LENGTH = 6;
-        JOB_NAME = REAL_SESSION.ISession.user.substring(0, JOB_LENGTH).toUpperCase() + "SF";
-        NON_HELD_JOBCLASS = TEST_ENVIRONMENT.systemTestProperties.zosjobs.jobclass;
+        // TODO: fix this test to run with any job name
+        JOB_NAME = "IEFBR14T"; //REAL_SESSION.ISession.user.substring(0, JOB_LENGTH).toUpperCase()
+        NON_HELD_JOBCLASS = defaultSystem.zosjobs.jobclass;
         SEARCH_STRING = "PGM=IEFBR14";
         REGEX_STRING = "IEFBR14|RC=0000";
         BAD_SEARCH_STRING = "bluhbluh";
 
     });
 
     afterAll(async () => {
-        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
+        // Retrieve jobs by prefix
+        const jobs = await GetJobs.getJobsByPrefix(REAL_SESSION, JOB_NAME);
+        testEnvironment.resources.jobs = jobs;
+
+        await TestEnvironment.cleanUp(testEnvironment);
     });
 
     describe("Successful response", () => {
-        it("should be able to search for a string in every spool file for a job", () => {
+        it("should be able to search for a string in every spool file for a job", async () => {
             const response = runCliScript(__dirname + "/__scripts__/job/search_string_spool_content.sh",
-                TEST_ENVIRONMENT, [IEFBR14_JOB, JOB_NAME, SEARCH_STRING]);
+                testEnvironment, [IEFBR14_JOB, JOB_NAME, SEARCH_STRING]);
+
             expect(response.stderr.toString()).toBe("");
             expect(response.status).toBe(0);
             expect(response.stdout.toString()).toContain(SEARCH_STRING);
         });
 
-        it("should be able to search for a regex in every spool file for a job", () => {
+        it("should be able to search for a regex in every spool file for a job", async () => {
             const argString = "--search-regex \"" + REGEX_STRING + "\"";
             const response = runCliScript(__dirname + "/__scripts__/job/search_no_job_submit.sh",
-                TEST_ENVIRONMENT, [JOB_NAME, argString]);
+                testEnvironment, [JOB_NAME, argString]);
+
+            // Add a delay to ensure job is recognized
+            await new Promise(resolve => setTimeout(resolve, 3000)); // Wait for 3 seconds
```

Comment:
Curious if we should use teh promisify(setTimeout) technique here??

Also... what about creating a wait/sleep meethog in the cli-test-utils, that way we have a single place where the `sleep` function is implemented (and if we decided the add logs, we only do it in one place) 😋

Reply:
yes good idea about making a sleep method